packages/common/src/store/purchase-content/sagas.ts

```diff
@@ -280,50 +372,74 @@ function* doStartPurchaseContentFlow({
     const extraAmountBN = new BN(extraAmount ?? 0).mul(BN_USDC_CENT_WEI)
     const totalAmountDueCentsBN = priceBN.add(extraAmountBN) as BNUSDC
 
-    if (purchaseMethod === PurchaseMethod.CARD) {
-      // Transition to 'buying USDC' stage
-      yield* put(buyUSDC())
-      yield* call(purchaseUSDC, { amount: totalAmountDueCentsBN })
-    }
-    // Manual transfer is a special case of existing balance. We expect the
-    // transfer to have occurred before this saga is invoked
-    else {
-      // No work needed here other than to check that the balance is sufficient
-      const balanceNeeded = getBalanceNeeded(
-        totalAmountDueCentsBN,
-        new BN(initialBalance.toString()) as BNUSDC,
-        usdcConfig.minUSDCPurchaseAmountCents
-      )
-      if (balanceNeeded.gtn(0)) {
+    const { blocknumber, splits } = yield* getPurchaseConfig({
+      contentId,
+      contentType
+    })
+    const balanceNeeded = getBalanceNeeded(
+      totalAmountDueCentsBN,
+      new BN(initialBalance.toString()) as BNUSDC,
+      usdcConfig.minUSDCPurchaseAmountCents
+    )
+
+    if (balanceNeeded.lten(0)) {
+      // No balance needed, perform the purchase right away
+      yield* call(purchaseContent, audiusBackendInstance, {
```

Comment:
https://linear.app/audius/issue/PAY-2259/[qa]-existing-balance-being-used-even-if-card-option-is-selected ? cc @raymondjacobson should we remove the balance check here?